infra/concourse/build/developer-tools/Makefile

```diff
@@ -0,0 +1,87 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+SHELL := /usr/bin/env bash # Make will use bash instead of sh
+TERRAFORM_VERSION := 0.12.5
+CLOUD_SDK_VERSION := 255.0.0
+GSUITE_PROVIDER_VERSION := 0.1.22
+TERRAFORM_DOCS_VERSION := 0.6.0
+RUBY_VERSION := 2.6.3
+
+DOCKER_TAG_VERSION_DEVELOPER_TOOLS := 0.0.1
+DOCKER_IMAGE_DEVELOPER_TOOLS := cft/developer-tools
+REGISTRY_URL := gcr.io/cloud-foundation-cicd
+
+
+.PHONY: build-image-developer-tools
+build-image-developer-tools:
+	docker build \
+		--build-arg CLOUD_SDK_VERSION=${CLOUD_SDK_VERSION} \
+		--build-arg GSUITE_PROVIDER_VERSION=${GSUITE_PROVIDER_VERSION} \
+		--build-arg TERRAFORM_VERSION=${TERRAFORM_VERSION} \
+		--build-arg TERRAFORM_DOCS_VERSION=${TERRAFORM_DOCS_VERSION} \
+		--build-arg RUBY_VERSION=${RUBY_VERSION} \
+		-t ${DOCKER_IMAGE_DEVELOPER_TOOLS}:${DOCKER_TAG_VERSION_DEVELOPER_TOOLS} .
+
+.PHONY: release-image-developer-tools
+release-image-developer-tools:
+	docker tag ${DOCKER_IMAGE_DEVELOPER_TOOLS}:${DOCKER_TAG_VERSION_DEVELOPER_TOOLS} \
+  ${REGISTRY_URL}/${DOCKER_IMAGE_DEVELOPER_TOOLS}:${DOCKER_TAG_VERSION_DEVELOPER_TOOLS}
+	docker push ${REGISTRY_URL}/${DOCKER_IMAGE_DEVELOPER_TOOLS}:${DOCKER_TAG_VERSION_DEVELOPER_TOOLS}
+
+# The docker_run task is provided as an example and reference for other CFT
+# Terraform modules to use.  Each module should have an identical `make
+# docker_run` task in the module's repository.
+.PHONY: docker_run
+docker_run: ## Launch a shell within the Docker test environment
+	docker run --rm -it \
+		-e SERVICE_ACCOUNT_JSON \
+		-v $(CURDIR):/workspace \
+		cft/developer-tools:0.0.1 \
+		/bin/bash
+
+.PHONY: docker_test_lint
+docker_test_lint:
+	docker run --rm -it \
+		-v $(CURDIR):/workspace \
+		cft/developer-tools:0.0.1 \
+		make test_lint
+
+.PHONY: test_lint
```

Comment:
Instead of copying "test_lint" into each repo we should provide a single command which can be invoked to do everything.

Reply:
How about this as the interface between a module and the developer-tools container image?

```
.PHONY: docker_test_lint
docker_test_lint:
	docker run --rm -it \
		-v $(CURDIR):/workspace \
		cft/developer-tools:0.0.1 \
		test_lint
```

If inside the container, the single command would be `test_lint`

Reply:
Yes that make sense.

Reply:
Updated